R/agg_scale.R

```diff
@@ -218,16 +219,20 @@ agg <- function(dt,
         overlapping_dt_severity = overlapping_dt_severity,
         include_missing = TRUE
       )
+      dt_intervals <- unique(dt[, .SD, .SDcols = cols])
     }
 
     # create one column to describe each interval
     gen_name(mapping, col_stem = col_stem, format = "interval")
     data.table::setnames(mapping, paste0(col_stem, "_name"), col_stem)
-    gen_name(dt, col_stem = col_stem, format = "interval")
```

Comment:
`gen_name` basically pastes together the start and end of interval columns to create a single name that describes the interval.

With large vectors `paste0` actually becomes pretty slow. The `profvis` example said this line was taking about 14 seconds (for ~14 million rows). Thats what motivated this switch to generating names for each unique interval and then taking advantage of data.table's quick joins.